xml/hwbp_registers.xml

```diff
@@ -1135,10 +1267,14 @@
             Data used together with \FcsrTextraThirtytwoMhselect.
         </field>
         <field name="mhselect" bits="25:23" access="WARL" reset="0">
-            0: Ignore \FcsrTextraThirtytwoMhvalue.
+            <value v="0" name="ignore">
+            Ignore \FcsrTextraThirtytwoMhvalue.
+            </value>
 
-            4: This trigger will only match if the low bits of
+            <value v="4" name="equal">
+            This trigger will only match if the low bits of
             \RcsrMcontext/\RcsrHcontext equal \FcsrTextraThirtytwoMhvalue.
+            </value>
 
             1, 5: This trigger will only match if the low bits of
```

Comment:
1,5,2,6 should have values.  That may require splitting them into separate entries.
- 1 matches when m/hcontext[0]=0 and m/hcontext[n:1]=mhvalue
- 5 matches when m/hcontext[0]=1 and m/hcontext[n:1]=mhvalue
- etc.

And maybe the names should be something like odd/even + context/vmid.

Tangent: In looking at this now, I don't exactly remember why we have value 4 plus 1 and 5.  Obviously 1/5 have an extra bit of range but why was that important?  In the mcontext CSR, it is recommended that the size be limited to 6 on RV32 and 13 on RV64 which means that we don't need the extra range.

Reply:
Hah. I dug up the original PR because I was puzzling about this as well. What happened is that this used to be a 1-bit field, and then you turned it into a 3-bit field with the chosen encodings to retain backwards compatibility.

Can I punt on this case for this change? I'll make up names so they'll show up in the PDF consistently.

Reply:
That sounds vaguely familiar.  I think that Greg came up with that idea.  Maybe it made sense to have an extra bit at the time, especially since there was going to be an extra bit for the VMID.  That doesn't completely make sense to me now but I don't think we should be changing that now.  Anyway...

It's OK to punt for now.  Get the 99% solution done and then incremental improvements can happen over time.

Reply:
It happened at https://github.com/riscv/riscv-debug-spec/pull/549/files#diff-979319471706782a119ea34ac26af26137b1beedfbeec7603a5a314f841810ebL828 if you're curious.

Reply:
More archaeology: Looks like it was Ernie's idea in the "Debug support for Hypervisor extension" email thread from "Sun, 2 Aug 2020 16:27:23 -0600" and I guess I wrote it up.